SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialComponentTest/SpatialComponentTest.cpp

```diff
@@ -201,41 +199,47 @@ void ASpatialComponentTest::CheckComponentsCrossServer(ASpatialComponentTestActo
 		{
 			if (LocalWorkerDefinition.Id == StartServerId)
 			{
-				RequireTrue(VerifyTestActorComponents(Actor, 3),
-							TEXT("Spawning server - OnActorReady component, OnAuthorityGained component and OnAuthorityLost component"));
+				VerifyTestActorComponents(
+					Actor, 3, TEXT("Spawning server - OnActorReady component, OnAuthorityGained component and OnAuthorityLost component"));
 				FinishStep();
 			}
 			else if (LocalWorkerDefinition.Id == EndServerId)
 			{
-				RequireTrue(VerifyTestActorComponents(Actor, 1), TEXT("Migrated server - OnAuthorityGained component"));
+				VerifyTestActorComponents(Actor, 1, TEXT("Migrated server - OnAuthorityGained component"));
 				FinishStep();
 			}
 		}
 		else // Support for Native / Single Worker.
 		{
-			RequireTrue(VerifyTestActorComponents(Actor, 2),
-						TEXT("Native / Single Worker - OnActorReady component and OnAuthorityGained component"));
+			VerifyTestActorComponents(Actor, 2, TEXT("Native / Single Worker - OnActorReady component and OnAuthorityGained component"));
 			FinishStep();
 		}
 	}
 	else // Clients
 	{
-		RequireTrue(VerifyTestActorComponents(Actor, 0), TEXT("Clients"));
+		VerifyTestActorComponents(Actor, 0, TEXT("Clients"));
 		FinishStep();
 	}
 }
 
-bool ASpatialComponentTest::VerifyTestActorComponents(ASpatialComponentTestActor* Actor, int ExpectedTestComponentCount)
+bool ASpatialComponentTest::VerifyTestActorComponents(ASpatialComponentTestActor* Actor, int ExpectedTestComponentCount,
+													  const FString& Message)
 {
-	if (!IsValid(Actor) || !Actor->HasActorBegunPlay())
+	if (!RequireTrue(IsValid(Actor), FString::Printf(TEXT("(%s) Actor is valid"), *Message)))
```

Comment:
I believe moving the assertion macros away from the call site is not the best practice since it makes it harder to spot the point of failure.

Reply:
I think in this specific case it's the lesser evil - there are three possible reasons this method can fail, and as they were compressed into one bool before, there was too little information to analyze test failures. I agree this isn't the best way possible.

Reply:
Not keen on before/after code to be honest.

Given the check for "validactor" and "hasbegunplay" is consistent across all error cases being handled, why isn't this done first (in the function above), and only once, leave the rest of the error handling checks to be only looking at the components?

Reply:
I'm not sure if moving stuff from VerifyTestActorComponents to CheckComponents and CheckComponentsCrossServer makes that big of a difference, considering how the test is structured.

Reply:
I think the clarity will be greatly improved, but I won't block the review because of this.